### PR TITLE
Add a few missing auth guards I had missed

### PR DESCRIPTION
**Pull request by Senjai**
Discovered I had missed these when working on a client implimentation

### CODE REVIEW

File: backend/app/views/spree/admin/promotion_categories/index.html.erb [hunk 1]
```diff
@@ -27,8 +29,12 @@
         <td><%= promotion_category.name %></td>
         <td><%= promotion_category.code%></td>
         <td class="actions">
-          <%= link_to_edit promotion_category, :no_text => true %>
-          <%= link_to_delete promotion_category, :no_text => true %>
+          <% if can?(:edit, promotion_category) %>
```
Comment (magnusvk) [inline backend/app/views/spree/admin/promotion_categories/index.html.erb]: I was confused that we use `:edit` here and `:update` below. Turns out those are aliases but consistency is nice so I'll nitpick here.

**Comment by magnusvk** [comment]
:+1:

File: backend/app/views/spree/admin/promotion_categories/index.html.erb (hunk 1 above)
Comment (Senjai) [inline backend/app/views/spree/admin/promotion_categories/index.html.erb]: Yes, you are correct. I'll change that.

**Comment by jhawthorn** [comment]
lgtm

**Comment by Senjai** [comment]
I'm not entirely sure of the circumstances around locking down the stock locations. If anyone has any comments regarding that please let me know.

File: backend/app/views/spree/admin/users/_form.html.erb [hunk 2]
```diff
@@ -26,17 +26,25 @@
...
-            <%= label_tag stock_location.name %>
-          </li>
-        <% end %>
-      </ul>
-    </div>
+    <% if can?(:display, Spree::StockLocation) %>
+      <div data-hook="admin_user_form_stock_locations" class="field">
+        <%= label_tag nil, Spree.t(:stock_locations) %>
+        <ul>
+          <% if can?(:associate_users, Spree::StockLocation) %>
```
Comment (magnusvk) [inline backend/app/views/spree/admin/users/_form.html.erb]: Chatting with @Senjai we think it'd be better to check against `can?(:manage, Spree::UserStockLocation)` here.

**Comment by magnusvk** [comment]
:+1: @jhawthorn want to take another look at this now that it's updated? I think it's ready to merge.

**Comment by jhawthorn** [comment]
:+1:

**Pull request merged**